### PR TITLE
SDK-2189: Add support for providing a flag to allow expired documents…

### PR DESCRIPTION
**Pull request by artem-yoti**
… at session creation

### CODE REVIEW

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentFilter.java [hunk 1]
```diff
@@ -24,4 +28,13 @@ public String getType() {
         return type;
     }
 
+    /**
+     * Whether allow expired documents
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentFilter.java:32]:
```suggestion
     * Returns if the {@link DocumentFilter} should allow expired documents
```

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilter.java [hunk 2]
```diff
@@ -39,34 +39,79 @@ public static class Builder {
...
+         *
+         * @return the builder
+         */
+        public Builder withAllowExpiredDocuments() {
+            this.allowExpiredDocuments = true;
+            return this;
+        }
+
+        /**
+         * Sets the flag whether allow expired documents or not
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilter.java:102]:
```suggestion
         * Sets the flag to allow/disallow expired documents
```

File: yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java [hunk 3]
```diff
@@ -92,4 +92,24 @@ public void shouldAcceptPreBuiltDocumentRestriction() {
...
+        DocumentRestrictionsFilter result = DocumentRestrictionsFilter.builder()
+                .forWhitelist()
+                .withAllowExpiredDocuments(false)
+                .build();
+
+        assertThat(result.getAllowExpiredDocuments(), is(false));
+    }
+
+    @Test
+    public void shouldSetAllowExpiredDocumentsFlagToTrueByDefault() {
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java:106]: The SDK's should _not_ provide default values, this is handled by our backend services.  Please use a `Boolean` object here, so the property is omitted when serializing to JSON so the backend handles default values.

File: yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilterTest.java [hunk 4]
```diff
@@ -79,4 +79,22 @@ public void shouldBuildBlacklistedDocumentTypeRestriction() {
...
+    public void shouldSetAllowExpiredDocumentsFlag() {
+        OrthogonalRestrictionsFilter result = OrthogonalRestrictionsFilter.builder()
+                .withAllowExpiredDocuments(false)
+                .build();
+
+        assertThat(result.getAllowExpiredDocuments(), is(false));
+    }
+
+    @Test
+    public void shouldSetAllowExpiredDocumentsFlagToTrueByDefault() {
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilterTest.java:92]: Same here, no default values please

**Review by MrBurtyyy — CHANGES_REQUESTED** [review]

File: yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java (hunk 3 above)
Comment (artem-yoti) [reply yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java:106]: I see...
Sorry, I was puzzled by the example
`"allow_expired_documents": true // THIS IS NEW, optional, default to true` in the epic. That's the requirement for our service, not for SDK

Fixed

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilter.java (hunk 2 above)
Comment (artem-yoti) [reply yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilter.java]: Changed

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentFilter.java (hunk 1 above)
Comment (artem-yoti) [reply yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/DocumentFilter.java]: Changed

**Review by MrBurtyyy — APPROVED** [review]

**Pull request merged**